VultisigApp/VultisigApp/Services/Thorchain/ThorchainService.swift

```diff
@@ -312,7 +312,118 @@ extension ThorchainService {
     }
 }
 
+// MARK: - RUJI Merge/Unmerge Functionality
+extension ThorchainService {
+    
+    /// Fetch merged RUJI balance for a specific token
+    /// - Parameters:
+    ///   - thorAddress: The THORChain address to query
+    ///   - tokenSymbol: The token symbol to check (e.g., "THOR.KUJI", "THOR.RKUJI")
+    /// - Returns: A tuple containing (ruji amount, shares, price per share)
+    func fetchRujiBalance(thorAddr: String, tokenSymbol: String) async throws -> (ruji: Decimal, shares: String, price: Decimal) {
```

Comment:
can you please define a struct for `(ruji: Decimal, shares: String, price: Decimal)`